tests/upgrades/test_repository.py

```diff
@@ -203,3 +203,82 @@ def test_post_scenario_custom_repo_check(self, target_sat, pre_upgrade_data):
         )[0]
         result = rhel_client.execute(f'yum install -y {FAKE_4_CUSTOM_PACKAGE_NAME}')
         assert result.status == 0
+
+    @pytest.mark.pre_upgrade
+    def test_pre_scenario_custom_repo_check_sca_toggle(
+        self, target_sat, sat_upgrade_chost, save_test_data, default_location
+    ):
+        """This is a pre-upgrade scenario test to verify that repositories in a non-sca org
+        set to "Enabled" should be overridden to "Enabled(Override)" when upgrading to 6.14.
+
+        :id: preupgrade-65e1e312-a743-4605-b226-f580f523377f
+
+        :steps:
+            1. Before Satellite upgrade.
+            2. Create new Organization, Location.
+            3. Create Product, Custom Repository, Content view.
+            4. Create Activation Key and add Subscription.
+            5. Create a Content Host, register it, and check Repository Sets for enabled Repository.
+
+        :expectedresults:
+
+            1. Custom Repository is created.
+            2. Custom Repository is enabled on Host.
+
+        """
+        org = target_sat.api.Organization().create()
+        lce = target_sat.api.LifecycleEnvironment(organization=org).create()
+
+        product = target_sat.api.Product(organization=org).create()
+        repo = target_sat.api.Repository(product=product.id, url=settings.repos.yum_1.url).create()
+        repo.sync()
+        content_view = target_sat.publish_content_view(org, repo)
+        content_view.version[0].promote(data={'environment_ids': lce.id})
+        ak = target_sat.api.ActivationKey(
+            content_view=content_view, organization=org.id, environment=lce
+        ).create()
+        if not target_sat.is_sca_mode_enabled(org.id):
```

Comment:
I think by default all org has `SCA` enabled, isnt it ?

if yes, what would be the else condition here?